examples/application/abi/poll/contract.py

```diff
@@ -0,0 +1,182 @@
+# This example is provided for informational purposes only and has not been audited for security.
+import json
+from typing import Literal
+from pyteal import *
+
+
+def on_delete() -> Expr:
+    return Assert(Txn.sender() == Global.creator_address())
+
+
+router = Router(
+    name="OpenPollingApp",
+    descr="A polling application with no restrictions on who can participate.",
+    bare_calls=BareCallActions(
+        delete_application=OnCompleteAction.call_only(on_delete())
+    ),
+)
+
+open_key = Bytes(b"open")
+resubmit_key = Bytes(b"resubmit")
+option_name_prefix = b"option_name_"
+option_name_keys = [
+    Bytes(option_name_prefix + b"\x00"),
+    Bytes(option_name_prefix + b"\x01"),
+    Bytes(option_name_prefix + b"\x02"),
+]
+option_count_prefix = b"option_count_"
+option_count_keys = [
+    Bytes(option_count_prefix + b"\x00"),
+    Bytes(option_count_prefix + b"\x01"),
+    Bytes(option_count_prefix + b"\x02"),
+]
+
+
+@router.method(no_op=CallConfig.CREATE)
+def create(
+    options: abi.StaticArray[abi.String, Literal[3]], can_resubmit: abi.Bool
+) -> Expr:
+    """Create a new polling application.
+
+    Args:
+        options: A list of options for the poll. This list should not contain duplicate entries.
+        can_resubmit: Whether this poll allows accounts to change their submissions or not.
+    """
+    return Seq(
+        App.globalPut(open_key, Int(0)),
+        App.globalPut(resubmit_key, can_resubmit.get()),
+        App.globalPut(option_name_keys[0], options[0].use(lambda value: value.get())),
```

Comment:
sanity checking that a more direct approach for setting the value isn't available